lightning/src/ln/channel.rs

```diff
@@ -4237,8 +4243,10 @@ impl<ChanSigner: ChannelKeys> Writeable for Channel<ChanSigner> {
 	}
 }
 
-impl<ChanSigner: ChannelKeys + Readable> Readable for Channel<ChanSigner> {
-	fn read<R : ::std::io::Read>(reader: &mut R) -> Result<Self, DecodeError> {
+const MAX_ALLOC_SIZE: usize = 64*1024;
```

Comment:
I know 64 kB is used across the codebase but what's the rational picking up this value ? Is it a upper bound promise made by `rustc` default memory allocator or an assumption on the underlying system ?

Reply:
Its random, basically. Its a nice round number that is small enough that its hard to add up enough 64KB chunks to cause damage but big enough that we basically never hit it. Anything a few x higher or lower would probably work just as well.